lib/active_merchant/billing/gateways/mondido.rb

```diff
@@ -0,0 +1,448 @@
+        # It’s good practice (required) in many juristictions not to take a payment from a
+        #   customer until the goods are shipped.
+
+        put = {}
+        add_amount(put, money, options)
+        commit(:put, "transactions/#{authorization}/capture", put)
+      end
+
+      def refund(money, authorization, options={})
+        # Refund money to a card.
+        # This may need to be specifically enabled on your account and may not be supported by all gateways
+
+        requires!(options, :reason)
+        post = {
+          # transaction_id  int *required
+          #   ID for the transaction to refund
+          :transaction_id => authorization.to_i,
+
+          # reason string *required
+          #   The reason for the refund. Ex. "Cancelled order"
+          :reason => options[:reason]
+        }
+
+        add_amount(post, money, options)
+        commit(:post, 'refunds', post)
+      end
+
+      def void(authorization, options={})
+        # Entirely void a transaction.
+        # Any amount into a refund will cancel the whole reservation
+
+        refund(100, authorization, options)
+      end
+
+      def verify(credit_card, options={})
+        # Test a payment authorizing a value of 1.00
+        # Then void the transaction and refund the value
+        options[:reason] ||= "Active Merchant Verify"
+
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def store(payment, options = {})
+        post = {
+          # currency  string* required
+          :currency => options[:currency] || self.default_currency,
+
+          # test bool
+          #   Must be true if you are using a test card number.
+          :test => test?
+        }
+
+        # customer_ref  string
+        #   Merchant specific customer ID.
+        #   If this customer exists the card will be added to that customer.
+        #   If it doesn't exists a customer will be created.
+        post.merge!( :customer_ref => options[:customer].to_s ) if options[:customer]
+
+        add_credit_card(post, payment)
+        commit(:post, 'stored_cards', post)
+      end
+
+      def unstore(id, options={})
+        params = {}
+        commit(:delete, "stored_cards/#{id}", params)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Basic )\w+), '\1[FILTERED]').
+          gsub(%r((card_holder=)\w+), '\1[FILTERED]').
+          gsub(%r((card_cvv=)\d+), '\1[FILTERED]').
+          gsub(%r((card_expiry=)\w+), '\1[FILTERED]').
+          gsub(%r((card_number=)\d+), '\1[FILTERED]').
+          gsub(%r((card_type=)\w+), '\1[FILTERED]').
+          gsub(%r((hash=)\w+), '\1[FILTERED]').
+          gsub(%r((amount=)\w+), '\1[FILTERED]')
```

Comment:
Alright! Removed from scrub :) - Thanks. I'm learning during this process hehe :D Seriously.